### PR TITLE
Cone sizeref fixes

### PR DESCRIPTION
**Pull request by etpinard**
fixes https://github.com/plotly/plotly.js/issues/2700 and https://github.com/plotly/plotly.js/issues/2701 (with the help https://github.com/gl-vis/gl-cone3d/pull/12).

This PR also improves hover on cones by picking the entire cone as opposed to overlaying `gl-scatter3d` points and using those for picking. Note that, the hover label remain at the same positions as before.

I still need to fixup a few tests, but I would appreciate a review of commit https://github.com/plotly/plotly.js/commit/4836861130a1a2944d42ca2fc2025efa2027d68d (where I tried to make sense of `sizemode: 'absolute'`) and commit https://github.com/gl-vis/gl-cone3d/pull/12/commits/3b3ef9d465a8e30d14a9a4a1cc88d4deebec64a9 (where I impleted a "local" find-vectorScale algo as proposed by @alexcjohnson  in https://github.com/gl-vis/gl-cone3d/pull/12/commits/3b3ef9d465a8e30d14a9a4a1cc88d4deebec64a9)

### CODE REVIEW

File: src/traces/cone/convert.js [hunk 1]
```diff
@@ -88,15 +87,21 @@ function convert(scene, trace) {
...
-    coneOpts[sizeMode2sizeKey[trace.sizemode]] = trace.sizeref;
     coneOpts.coneOffset = anchor2coneOffset[trace.anchor];
 
-    var meshData = conePlot(coneOpts);
+    if(trace.sizemode === 'scaled') {
+        // unitless sizeref
+        coneOpts.coneSize = trace.sizeref || 0.5;
+    } else {
+        // sizeref here has unit of velocity
+        coneOpts.coneSize = (trace.sizeref / trace._normMax) || 0.5;
```
Comment (etpinard) [inline src/traces/cone/convert.js]: Previously, `coneSize` under `sizemode: 'absolute'` was scaled using the max norm of the "scaled" u/v/w vectors (that same `dataScale` than in https://github.com/plotly/plotly.js/issues/2646). This made `sizeref` under `sizemode: 'absolute'` had little meaning. 

Now, `sizeref` under `sizemode: 'absolute' is scaled by the "input" u/v/w max norm should make this thing a little less confusing.

File: src/traces/cone/attributes.js [hunk 2]
```diff
@@ -112,20 +112,26 @@ var attrs = {
...
         dflt: 'scaled',
         description: [
-            'Sets the mode by which the cones are sized.',
-            'If *scaled*, `sizeref` scales such that the reference cone size',
-            'for the maximum vector magnitude is 1.',
-            'If *absolute*, `sizeref` scales such that the reference cone size',
-            'for vector magnitude 1 is one grid unit.'
+            'Determines whether `sizeref` is set as a *scaled* (i.e unitless) scalar',
+            '(normalized by the max u/v/w norm in the vector field) or as',
+            '*absolute* value (in unit of velocity).'
```
Comment (alexcjohnson) [inline src/traces/cone/attributes.js]: I wouldn't use the word "velocity" here - you can show many other quantities as vector fields (electric fields, as just one common example). Just say "in the same units as the vector field" or some such.

File: src/traces/cone/attributes.js [hunk 3]
```diff
@@ -112,20 +112,26 @@ var attrs = {
...
         role: 'info',
         editType: 'calc',
         min: 0,
-        dflt: 1,
-        description: 'Sets the cone size reference value.'
+        description: [
+            'Adjusts the cone size scaling.',
+            'The size of the cones is determined by their u/v/w norm multiplied a factor and `sizeref`.',
+            'This factor (computed internally) corresponds to the minimum "time" to travel across two',
+            'two successive x/y/z positions at the average velocity of those two successive positions',
```
Comment (alexcjohnson) [inline src/traces/cone/attributes.js]: two twos is one two too many

File: src/traces/cone/convert.js (hunk 1 above)
Comment (etpinard) [reply src/traces/cone/convert.js]: > What if the field is uniformly zero? 

Oh right, `Infinity` is truthy. Good call!

Comment (etpinard) [reply src/traces/cone/convert.js]: > When is the || 0.5 (in either case) necessary?

Currently, the `sizeref` attribute declaration does not set a `dflt` field. In practice I guess it does have a hard default of `0.5`. But technically under the `sizemode: 'absolute'`, the sizemode default is `0.5 * trace._normMax`.

**Comment by alexcjohnson** [comment]
Great! Smoothing off a few more rough edges with cones. 💃

**Pull request merged**